components/widgets/menu/menu_items/menu_top_notification.tsx

```diff
@@ -2,15 +2,34 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useSelector} from 'react-redux';
+
+import UpgradeLink from 'components/widgets/links/upgrade_link';
+
+import {GlobalState} from 'mattermost-redux/types/store';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+
+import {getRemainingDaysFromFutureTimestamp} from 'utils/utils.jsx';
+import {TrialPeriodDays} from 'utils/constants';
 
 import './menu_item.scss';
 
 type MenuTopNotificationProps = {
-    show: boolean;
-    id?: string;
-    children?: React.ReactNode;
+    id: string;
 }
-const MenuTopNotification: React.FC<MenuTopNotificationProps> = ({id, show, children}: MenuTopNotificationProps) => {
+const MenuTopNotification: React.FC<MenuTopNotificationProps> = ({id}: MenuTopNotificationProps) => {
+    const subscription = useSelector((state: GlobalState) => state.entities.cloud.subscription);
+    const license = useSelector(getLicense);
+
+    const isCloud = license.Cloud === 'true';
```

Comment:
We should add a null check here just in case `license` doesn't exist
```suggestion
    const isCloud = license?.Cloud === 'true';
```